packages/zql/src/builder/builder.ts

```diff
@@ -200,68 +200,57 @@ function buildPipelineInternal(
 function applyWhere(
   input: Input,
   condition: Condition,
-  // Remove `appliedFilter`
-  // Each branch can `fetch` with different filters from the same source.
-  // Or we do the union of queries approach and retain this `appliedFilters` and `sourceConnect` behavior.
-  // Downside of that being unbounded memory usage.
-  appliedFilters: boolean,
   delegate: BuilderDelegate,
 ): Input {
   switch (condition.type) {
     case 'and':
-      return applyAnd(input, condition, appliedFilters, delegate);
+      return applyAnd(input, condition, delegate);
     case 'or':
-      return applyOr(input, condition, appliedFilters, delegate);
+      return applyOr(input, condition, delegate);
     case 'correlatedSubquery':
       return applyCorrelatedSubqueryCondition(input, condition, delegate);
     case 'simple':
-      return applySimpleCondition(input, condition, appliedFilters);
+      return applySimpleCondition(input, condition);
   }
 }
 
 function applyAnd(
   input: Input,
   condition: Conjunction,
-  appliedFilters: boolean,
   delegate: BuilderDelegate,
 ) {
   for (const subCondition of condition.conditions) {
-    input = applyWhere(input, subCondition, appliedFilters, delegate);
+    input = applyWhere(input, subCondition, delegate);
   }
   return input;
 }
 
 export function applyOr(
   input: Input,
   condition: Disjunction,
-  appliedFilters: boolean,
   delegate: BuilderDelegate,
 ): Input {
   const [subqueryConditions, otherConditions] =
     groupSubqueryConditions(condition);
   // if there are no subquery conditions, no fan-in / fan-out is needed
   if (subqueryConditions.length === 0) {
-    return otherConditions.length > 0
-      ? new Filter(
-          input,
-          appliedFilters ? 'push-only' : 'all',
-          createPredicate({
-            type: 'or',
-            conditions: otherConditions,
-          }),
-        )
-      : new FanIn(new FanOut(input), []);
```

Comment:
this was producing a pipeline that was not fully connected.  not sure if this was intentional or not

Reply:
Got it, the filtering at the source ensures empty or is always empty now.